backend/src/api/album/get_filters.rs

```diff
@@ -0,0 +1,238 @@
+pub(super) struct AvailableAlbumFilters {
+    authors: Vec<String>,
+    timeframes: Vec<Timeframe>,
+    has_drafts: bool,
+}
+
+// This function returns available filters.
+// If filters are sent in it returns the available filters that can be used alternatively.
+// As an example, the returned authors are all authors that can be found
+// with all other filters applied. If the author is filtered on that is ignored for the
+// returned author since otherwise it would only return one value.
+// The same logic applies to all other filters.
+pub(super) async fn get<D: SqliteDatabase>(
+    Authorize(username): Authorize,
+    Query(filter): Query<AlbumFilters>,
+    Extension(state): Extension<Arc<AppState<D>>>,
+) -> Result<Json<AvailableAlbumFilters>, Error> {
+    let conn = state.pool.get().await.context("Failed to get connection")?;
+
+    conn.interact(move |conn| {
+        let mut authors: Vec<String> = get_filtered_values(
+            conn,
+            "author",
+            AlbumFilters {
+                author: None,
+                ..filter
+            },
+            username.clone(),
+        )?;
+
+        authors.dedup();
+
+        let mut timeframes: Vec<Timeframe> = get_filtered_values(
+            conn,
+            "timeframe_from AS \"from\", timeframe_to AS \"to\"",
+            AlbumFilters {
+                from: None,
+                to: None,
+                ..filter.clone()
+            },
+            username.clone(),
+        )?;
+
+        timeframes.dedup();
+
+        let drafts: Vec<i64> = get_filtered_values(
+            conn,
+            "1",
+            AlbumFilters {
+                draft: true,
+                ..filter
+            },
+            username,
+        )?;
+
+        Ok(Json(AvailableAlbumFilters {
+            authors,
+            timeframes,
+            has_drafts: !drafts.is_empty(),
+        }))
+    })
+    .await
+}
+
+fn get_filtered_values<T: DeserializeOwned>(
+    conn: &rusqlite::Connection,
+    column: &str,
+    filters: AlbumFilters,
+    username: String,
+) -> Result<Vec<T>, Error> {
+    let mut query = format!("SELECT {column} FROM albums");
+
+    let mut params: Vec<Box<dyn rusqlite::ToSql>> = Vec::new();
+
+    crate::api::album::apply_filters(&mut query, &mut params, filters, username);
+
+    let mut stmt = conn
+        .prepare(&query)
+        .context("Failed to prepare statement for draft query")?;
+
+    let output: Vec<T> = stmt
+        .query_map(rusqlite::params_from_iter(params.iter()), |row| {
+            Ok(from_row::<T>(row).unwrap())
+        })
+        .context("Failed to query authors")?
+        .try_collect()
+        .context("Failed to collect authors")?;
+
+    Ok(output)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::api::album::InsertAlbum;
+    use crate::util::test::{insert_album, insert_image, insert_user};
+    use assert_matches::assert_matches;
+
+    #[tokio::test]
+    async fn get_all_filters() {
```

Comment:
Feels like you're missing quite a bit of coverage with tests, but I still don't really understand the feature. Up to you

Reply:
I'm just too lazy to write nicer tests currently, I may come back and improve them later.